helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -156,12 +153,12 @@ void release(AssignableReplica assignableReplica) throws IllegalArgumentExceptio
         .forEach(entry -> updateCapacityAndUtilization(entry.getKey(), -1 * entry.getValue()));
   }
 
-  public Map<String, Set<String>> getCurrentAssignmentsMap() {
+  public Map<String, Map<String, String>> getCurrentAssignmentsMap() {
     return _currentAssignments;
   }
 
-  public Set<String> getCurrentAssignmentsByResource(String resource) {
-    return _currentAssignments.getOrDefault(resource, Collections.emptySet());
+  public Map<String, String> getCurrentAssignmentsByResource(String resource) {
```

Comment:
Could we please annotate the public methods, and especially makes sure the return values are marked with what the strings in the map stand for? It's not so easy for someone like me to decode this :)